### PR TITLE
Fix ArrayDelete bug

### PR DESCRIPTION
**Pull request by sanmut**
Fix ArrayDelete bug due to which more than the requested number of elements were getting deleted. Add more tests for 1D and 2D arrays for different values of index and length.

### CODE REVIEW

File: source/core/Array.cpp [hunk 1]
```diff
@@ -1243,7 +1243,7 @@ VIREO_FUNCTION_SIGNATURE7(ArrayDeleteND, TypedArrayCoreRef, StaticType, void,
 
     IntIndex startIndex = offset > 0 ? offset : 0;
     IntIndex endIndex = (offset +
```
Comment (mehagar) [inline source/core/Array.cpp]: Could you break apart this statement into variables so it's easier to read?

File: test-it/testList.json [hunk 2]
```diff
@@ -326,6 +326,8 @@
                 "SubVISimple.via",
                 "TemplateBasics.via",
                 "TestVI_ArrayDelete.via",
+                "TestVI_ArrayDelete_2.via",
```
Comment (cglzaguilar) [inline test-it/testList.json]: Please rename the test to provide more context as to what is being tested. 2 and 3 doesn't ring a bell.

File: test-it/ExpectedResults/TestVI_ArrayDelete_2.vtr [hunk 3]
```diff
@@ -0,0 +1,6 @@
+TestResult_6 true
```
Comment (cglzaguilar) [inline test-it/ExpectedResults/TestVI_ArrayDelete_2.vtr]: Please provide more context related to the case being exercised. Numbers 1 to 6 doesn't ring a bell.

File: source/core/Array.cpp (hunk 1 above)
Comment (sanmut) [reply source/core/Array.cpp]: Done.

File: test-it/ExpectedResults/TestVI_ArrayDelete_2.vtr (hunk 3 above)
Comment (sanmut) [reply test-it/ExpectedResults/TestVI_ArrayDelete_2.vtr]: Done

**Pull request merged**